### PR TITLE
Update AWS-SDK to v3

### PR DESCRIPTION
**Pull request by tristanmorgan**
This PR updates the Gem to use the aws-sdk-iam and aws-sdk-ec2 gems rather than load the whole SDK.

### CODE REVIEW

**Comment by tristanmorgan** [comment]
I've hit a deprecation warning. Working on it still.

**Comment by lamont-granquist** [comment]
looks like aws-sdk-v3 conflicts strictly with aws-sdk-v2 and cannot be bundled in the same gemfile.  that's going to block merging this until chef-provisioning-aws also gets merged:

```
Bundler could not find compatible versions for gem "aws-sdk-core":
  In Gemfile:
    kitchen-ec2 was resolved to 1.3.2, which depends on
      aws-sdk-ec2 was resolved to 1.0.0, which depends on
        aws-sdk-core (~> 3)

    chef-provisioning-aws (>= 2.0) was resolved to 2.0.0, which depends on
      aws-sdk (< 3.0, >= 2.1.26) was resolved to 2.1.26, which depends on
        aws-sdk-resources (= 2.1.26) was resolved to 2.1.26, which depends on
          aws-sdk-core (= 2.1.26)
```

there's also a bit of a concern that i've got which is that this is only the conflict that occurs in the gems that ship in the ChefDK by default, and the whole ecosystem looks like its going to need to take a flag day to upgrade to aws-sdk v3, unless i've missed something here.

**Comment by tristanmorgan** [comment]
OK, well until ChefDK gets the version bump we know its a simple change to update.

**Comment by arothian** [comment]
@lamont-granquist Any idea on when this might get done? Running into issues bundling with other gems that have migrated to sdk 3.

**Comment by tristanmorgan** [comment]
It seems Chef/Train also introduced a dependancy on the old V2 SDK too. https://github.com/chef/train/pull/239

File: kitchen-ec2.gemspec [hunk 1]
```diff
@@ -23,7 +23,8 @@ Gem::Specification.new do |gem|
   gem.add_dependency "test-kitchen", "~> 1.4", ">= 1.4.1"
   gem.add_dependency "excon"
   gem.add_dependency "multi_json"
-  gem.add_dependency "aws-sdk", "~> 2"
+  gem.add_dependency "aws-sdk-iam"
```
Comment (grosser) [inline kitchen-ec2.gemspec]: prefer to depend on a major version like `~> 1.2` so we don't bump across major versions would testing them

**Comment by grosser** [comment]
is there any issue open to get ChefDK updated ?

**Comment by coderanger** [comment]
@grosser Not currently, there is not currently a maintainer for chef-provisioning-aws which complicates things.

**Comment by tristanmorgan** [comment]
Unfortunately chef/train also introduced a dependancy on the old AWS-SDK v2.

**Comment by tas50** [comment]
This is very much something we want to ship so we can slim down the size of ChefDK, but there's additional work outside of kitchen-ec2 we have to do:

- Update chef-provisioning-aws
- Update InSpec

It's going to happen, but I don't have a timeline.

**Comment by tas50** [comment]
Closing this out in favor of #419

**Pull request closed without merging**